source/Camera/Basic/C++Halcon/CaptureViaZivid/CaptureViaZivid.cpp

```diff
@@ -19,50 +19,70 @@ HalconCpp::HObjectModel3D zividToHalconPointCloud(const Zivid::PointCloud &point
 	const auto width = pointCloud.width();
 	const auto height = pointCloud.height();
 
-	std::vector<float> pointsX(width * height);
-	std::vector<float> pointsY(width * height);
-	std::vector<float> pointsZ(width * height);
-	std::vector<float> colorsR(width * height);
-	std::vector<float> colorsG(width * height);
-	std::vector<float> colorsB(width * height);
-	std::vector<int> rowMapping(width*height);
-	std::vector<int> colMapping(width*height);
-
-	int numValidPoints = 0;
-	for (size_t i = 0; i < width*height; ++i)
+	int numberOfValidPoints = std::count_if(pointCloud.dataPtr(), pointCloud.dataPtr() + pointCloud.size(), [](Zivid::Point point)
```

Comment:
Didn't we find a way to not copy and pass by reference here?

Reply:
yes, by adding '&' before point in Zivid::Point point. I updated this right before our meeting with Eskil, that's why it is not updated here